pkg/expr/types/types.go

```diff
@@ -269,8 +269,8 @@ func MarshalRaw(results []*MetricData) []byte {
 // Consolidate returns a consolidated copy of this MetricData.
 func (r *MetricData) Consolidate(valuesPerPoint int) *MetricData {
 	ret := *r
-	if valuesPerPoint == 1 || valuesPerPoint == 0 {
-		ret.ValuesPerPoint = 1
+	if valuesPerPoint == 1 || valuesPerPoint <= 0 {
+		ret.ValuesPerPoint = valuesPerPoint
```

Comment:
Could you please elaborate why this line is changed?
If I understand correctly, this means that if `valuesPerPoint` is negative, put it in `ret.ValuesPerPoint` which doesn't seem correct. Also, I think if the panic observed is because of `maxDataPoints` being negative, we'll never reach this line because of the new change with `maxDataPoints > 0` condition on http_handlers.go file, right?

Reply:
Please check my latest commit, @emadolsky